### PR TITLE
1720 bugfix asset update action serialisation bug

### LINKED ISSUE #1720: ChangeAssetName with assetKey doesn't work.
I have a `ChangeAssetName` update action, created as follows:
````java
ChangeAssetName.ofAssetKeyAndVariantId(1, "assetKey", LocalizedString.ofEnglish("newAssetName"), true));
````
Now this update action, in JSON, looks as follows:
````JSON
    {
      "action": "changeAssetName",
      "staged": true,
      "variantId": 1,
      "name": {
        "en": "newAssetName"
      },
      "assetKey": "assetKey"
    }
````
Which should be valid to update the asset on our API.

However, issuing such an update action using the JVM SDK, leads to the following exception:

````
io.sphere.sdk.client.ErrorResponseException: detailMessage: Either 'assetId' or 'assetKey' must be defined
summary: POST https://api.sphere.io/java-sync-target-dev1/products/f374e87c-09f0-4888-8f06-6e58110a2ba7 failed  with response code 400 with X-Correlation-ID `java-sync-target-dev1/dfcd3a71-cd47-4bdb-9570-754f7c6ba964` on 2018-02-28T10:54:35.845Z
http response formatted body: {
  "statusCode" : 400,
  "message" : "Either 'assetId' or 'assetKey' must be defined",
  "errors" : [ {
    "code" : "InvalidInput",
    "message" : "Either 'assetId' or 'assetKey' must be defined",
    "action" : {
      "action" : "changeAssetName",
      "variantId" : 1,
      "staged" : true,
      "name" : {
        "en" : "newAssetName"
      }
    },
    "actionIndex" : 1
  } ]
}
request formatted body: {
  "version" : 1,
  "actions" : [ {
    "action" : "changeAssetName",
    "staged" : true,
    "variantId" : 1,
    "name" : {
      "en" : "newAssetName"
    }
  }]
}
additional notes: []
Javadoc: http://commercetools.github.io/commercetools-jvm-sdk/apidocs/io/sphere/sdk/client/ErrorResponseException.html

````

I debugged why this happens in the JVM SDK, apparently it's in the `MetaModelUpdateCommandDslImpl#httpRequestIntent` where the POJO update actions are parsed into JSON, specifically in this line 

````java
toJsonString(new UpdateCommandBody<>(getVersioned().getVersion(), getUpdateActions()))
````  

This actually leads to the aforementioned update action to be serialised to: 
````JSON
    {
      "action": "changeAssetName",
      "staged": true,
      "variantId": 1,
      "name": {
        "en": "newAssetName"
      }
    }
````

As you can see without the key which leads to the error from the platform.

### PR DESCRIPTION
**Pull request by heshamMassoud**
# Description

Adds getters (`getAssetKey`) to `ChangeAssetName` and `SetAssetCustomField` update actions to fix #1720.

Closes #1720 

# Checklist

- [ ] Update release Notes
- [ ] Add to the current github milestone

### CODE REVIEW

File: commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.products.Product;
+import org.junit.Test;
+
+import static io.sphere.sdk.models.LocalizedString.ofEnglish;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ChangeAssetNameTest {
```
Comment (katmatt) [inline commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java]: We normally focus on integration tests (in this case they're located in `ProductUpdateCommandIntegrationTest`) and don't write a lot of unit tests for our update actions. And I really like this test 👍 🥇

Comment (heshamMassoud) [reply commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java]: @katmatt oh, thanks for informing me about `ProductUpdateCommandIntegrationTest`. Didn't know about it. I will add some ITs there then ;)

Comment (heshamMassoud) [reply commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java]: @katmatt [Integration tests](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcb) added for:
1. [changeAssetNameByVariantIdAndAssetKey](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2380)
2. [changeAssetNameByVariantIdAndAssetKeyWithStaged](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2400)
3. [setAssetCustomTypeByVariantIdAndAssetKey](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2819)
4. [setAssetCustomTypeByVariantIdAndAssetKeyWithStaged](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2845)
3. [setAssetCustomFieldByVariantIdAndAssetKey](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2877)
4. [setAssetCustomFieldByVariantIdAndAssetKeyWithStaged](https://github.com/commercetools/commercetools-jvm-sdk/pull/1725/files#diff-bf86d04f48ca49a1342702ccf6779fcbR2916)

Comment (heshamMassoud) [reply commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java]: Thanks @katmatt :) would be glad if it's merged asap.

Comment (katmatt) [reply commercetools-models/src/test/java/io/sphere/sdk/products/commands/updateactions/ChangeAssetNameTest.java]: Ok and then we will ship it with our next release planned for this thursday.

**Pull request merged**